### PR TITLE
feat: Implement gene search functionality and gene information page (#11)

### PR DESCRIPTION
**Pull request by gromdimon**
### Tasks:
- [x] Implement home page with search bar
- [x] Implement Gene Details Page
- [x] Write tests

### CODE REVIEW

File: backend/app/main.py [hunk 1]
```diff
@@ -52,12 +52,16 @@ async def reverse_proxy(request: Request) -> Response:
...
-        backend_url = BACKEND_PREFIX_ANNONARS + url.path.replace("/proxy/annonars", "/annos")
+        backend_url = BACKEND_PREFIX_ANNONARS + url.path.replace("/proxy/annonars", "")
     elif url.path.startswith("/proxy/mehari"):
         backend_url = BACKEND_PREFIX_MEHARI + url.path.replace("/proxy/mehari", "")
     elif url.path.startswith("/proxy/viguno"):
         backend_url = BACKEND_PREFIX_VIGUNO + url.path.replace("/proxy/viguno", "")
 
+    import sys
+
+    print(f"backend_url = {backend_url}", file=sys.stderr)
```
Comment (holtgrewe) [inline backend/app/main.py:63]: remove

File: backend/app/main.py [hunk 2]
```diff
@@ -82,6 +86,14 @@ async def reverse_proxy(request: Request) -> Response:
...
 
 # Routes
+@app.get("/api/search")
+async def search_gene(
+    geneSymbol: str = Query(...), spdi: str = Query(""), genomeRelease: str = Query("hg19")
+):
+    gene_details = {"geneSymbol": geneSymbol, "genomeRelease": genomeRelease}
+    return JSONResponse(content=gene_details)
+
+
```
Comment (holtgrewe) [inline backend/app/main.py:96]: remove?

File: frontend/package.json [hunk 3]
```diff
@@ -14,9 +14,13 @@
     "format:check": "prettier --list-different src/ *.ts *.json"
   },
   "dependencies": {
+    "@mdi/font": "^7.2.96",
+    "@mdi/js": "^7.2.96",
+    "axios": "^1.4.0",
```
Comment (holtgrewe) [inline frontend/package.json:19]: we don't want this

File: frontend/src/api/annonars.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+const API_BASE_URL =
+  import.meta.env.MODE == 'development' ? '//localhost:8080/proxy/annonars' : '/proxy/annonars'
```
Comment (holtgrewe) [inline frontend/src/api/annonars.ts:2]: Maybe we should an `export const API_BASE_PREFIX` in a more central module (`api/common.ts`?) and import this from there. then use `const API_BASE_URL = \`${API_BASE_PREFIX}/\`` here?

File: frontend/src/components/HeaderDetailPage.vue [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+} else {
+  // const searchTerm = geneInfoStore.geneInfo.searchTerm
+}
+</script>
+
+<template>
+  <v-app-bar app class="top-bar">
+    <v-toolbar-title>
+      <router-link to="/">
+        <img src="@/assets/reev-logo.png" id="logo" alt="logo" width="100" />
```
Comment (holtgrewe) [inline frontend/src/components/HeaderDetailPage.vue:20]: Can't we just use the SVG here?

File: frontend/src/stores/geneInfo.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
+/** Store for gene details.
+ *
+ * This includes the data retrieved from the APIs.
+ *
```
Comment (holtgrewe) [inline frontend/src/stores/geneInfo.ts:4]: Remove such trailing comment lines.

File: frontend/src/views/AboutView.vue [hunk 7]
```diff
@@ -1,29 +1,42 @@
-<script lang="ts" setup>
-import { useCounterStore } from '../stores/counter'
-
-const counterStore = useCounterStore()
+<script setup lang="ts">
+import Header from '@/components/HeaderDefault.vue'
```
Comment (holtgrewe) [inline frontend/src/views/AboutView.vue:2]: Should this be `import HeaderDefault`?

File: frontend/src/views/ContactView.vue [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+  <v-container class="contact-view">
+    <v-row>
+      <v-col cols="12">
+        <h2 class="title">Contact Us</h2>
+        <p class="description">
+          We'd love to hear from you! Feel free to reach out to us through any of the following
+          channels:
+        </p>
+        <div class="contact-links">
+          <v-btn href="https://github.com/bihealth" target="_blank">
```
Comment (holtgrewe) [inline frontend/src/views/ContactView.vue:16]: Should be `https://github.com/bihealth/reev/discussions`

File: frontend/src/views/ContactView.vue [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+            <v-icon left>mdi-github</v-icon>
+            GitHub
+          </v-btn>
+          <v-btn href="mailto:random.mail@adress.com" target="_blank">
```
Comment (holtgrewe) [inline frontend/src/views/ContactView.vue:20]: should be cubi-helpdesk@bih-charite.de and we don't need target blank here.

File: frontend/src/views/HomeView.vue [hunk 10]
```diff
@@ -1,9 +1,105 @@
...
+
+import { ref } from 'vue'
+import { useRouter } from 'vue-router'
+// import { useGeneInfoStore } from '@/stores/geneInfo'
+
+import Header from '@/components/HeaderDefault.vue'
+import SearchBar from '@/components/SearchBar.vue'
+
+const router = useRouter()
+// const geneInfoStore = useGeneInfoStore()
```
Comment (holtgrewe) [inline frontend/src/views/HomeView.vue:16]: Remove such commented out lines.

File: frontend/tsconfig.app.json [hunk 11]
```diff
@@ -1,9 +1,19 @@
...
+    "src/**/*.js",
+    "src/**/*.jsx",
+    "src/**/*.json",
+    "src/**/*.d.ts",
+    "src/api/utils.ts"
+  ],
   "exclude": ["src/**/__tests__/*"],
   "compilerOptions": {
-    "composite": true,
+    // "composite": true,
```
Comment (holtgrewe) [inline frontend/tsconfig.app.json:16]: remove line?

File: info?hgnc-id=HGNC:1100
Comment (holtgrewe) [inline info?hgnc-id=HGNC:1100:1]: Should this be removed?

File: package-lock.json
Comment (holtgrewe) [inline package-lock.json:1]: I don't think that we want to have this at the top level. I'ts probably an artifact from running `npm install` in the wrong place.

File: frontend/src/views/GeneDetailView.vue [hunk 12]
```diff
@@ -0,0 +1,596 @@
...
+// TODO: There is no variantQueryStore at the moment, so
+// this function is modifyed from the original.
+const linkOutPubMedHpoTerms = computed((): string | null => {
+  if (geneInfoStore.geneInfo?.hgnc?.symbol) {
+    const symbol = geneInfoStore.geneInfo.hgnc.symbol
+    return `https://www.ncbi.nlm.nih.gov/pubmed/?term=${symbol}`
+  } else {
+    return null
+  }
+})
```
Comment (holtgrewe) [inline frontend/src/views/GeneDetailView.vue:38]: Remove this and the corresponding linkout as well as the stop words ;-)

**Review by holtgrewe — CHANGES_REQUESTED** [review]

File: frontend/src/api/common.ts [hunk 13]
```diff
@@ -0,0 +1,4 @@
+const API_BASE_PREFIX =
+  import.meta.env.MODE == 'development' ? '//localhost:8080/proxy/annonars' : '/proxy/annonars'
+
```
Comment (holtgrewe) [inline frontend/src/api/common.ts:3]: rather use `export const`?

File: frontend/src/views/HomeView.vue [hunk 14]
```diff
@@ -1,9 +1,103 @@
...
+interface RouteLocationFragment {
+  name: string
+  params?: any
+}
+
+type RouteLoctionBuilder = () => RouteLocationFragment
+
+// We iterate the regexps in the `Map` and will use the route from the
+// first match.
+const searchRegexp: [RegExp, RouteLoctionBuilder][] = [
```
Comment (holtgrewe) [inline frontend/src/views/HomeView.vue:41]: Should probably be called `SEARCH_REGEXPS`.

File: frontend/tsconfig.app.json [hunk 15]
```diff
@@ -1,9 +1,18 @@
...
+  "include": [
+    "env.d.ts",
+    "src/**/*.ts",
+    "src/**/*.tsx",
+    "src/**/*.vue",
+    "src/**/*.js",
+    "src/**/*.jsx",
+    "src/**/*.json",
+    "src/**/*.d.ts",
+    "src/api/utils.ts"
```
Comment (holtgrewe) [inline frontend/tsconfig.app.json:12]: Is this needed?

File: frontend/src/views/GeneDetailView.vue [hunk 16]
```diff
@@ -0,0 +1,548 @@
...
+// When the component is mounted or the search term is changed through
+// the router then we need to fetch the gene information from the backend
+// through the store.
+
+onMounted(loadDataToStore)
+
+watch(() => props.searchTerm, loadDataToStore)
+watch(() => route.hash, scrollToSection)
+
+const sections = [
```
Comment (holtgrewe) [inline frontend/src/views/GeneDetailView.vue:43]: Name `SECTIONS`?

**Review by holtgrewe — COMMENTED** [review]
Generally, looks good to me, except for small adjustments.

Next step is to add tests?

**Comment by gromdimon** [comment]
### Some notes:
- We ommit tests in docker builds : see .dockerignore
- `AnnonarsClient` does not properly handles exceptions
- SearchBar selection for genomeRelease is not tested
- HomeView `performSearch` is not tested
- GeneDetailView navigation bar redirections with `#` in url is not tested

**Pull request merged**